src/conn.c

```diff
@@ -4030,6 +4031,43 @@ natsConnection_GetLastError(natsConnection *nc, const char **lastError)
     return s;
 }
 
+natsStatus
+natsConnection_ReadLastError(natsConnection *nc, char *buf, size_t n)
+{
+    natsStatus  s;
+
+    if (nc == NULL)
+        return nats_setDefaultError(NATS_INVALID_ARG);
+
+    natsConn_Lock(nc);
+
+    s = nc->err;
+    if (s == NATS_OK)
+        nc->errStr[0] = '\0';
+    else if (nc->errStr[0] == '\0')
+        snprintf(nc->errStr, sizeof(nc->errStr), "%s", natsStatus_GetText(s));
+
+    if ((buf != NULL) && (n > 0))
+    {
+        size_t errLen = strlen(nc->errStr);
+        strncpy(buf, nc->errStr, n-1);
+        buf[n-1] = '\0';
+
+        bool truncate = ((errLen > (n-1)) && (n > 4));
```

Comment:
Do we really need to do that (add the `...`)? If this function replaces (for data race reasons) the `natsConnection_GetLastError()`, then we were not doing that there, so I would not do it here.

And if you decide that it is not worth it at the end, then maybe simply replace the copy with a snprintf.

Reply:
We don't... but since we don't give out the size of the full text, I feel like we should at least provide a visual indication of truncation when it happens. Yes, I'll add a test.